### ISSUE TITLE
rdoc and json are incompatible

### ISSUE THREAD

**Issue by isuruf** [SciRuby/nmatrix#553]
`rdoc` needs `json ~> 1.4` (>=1.4, < 2.0) and nmatrix requires >= 2.0.1.

**Comment by mohawkjohn** [comment]
I believe I just put in ~> 2.0.1 because that's the one I got working at the time. Do you want to see what happens in your PR if you change it to 1.4?

**Comment by isuruf** [comment]
I just did, and it fails to install json 1.8.3. json needs to be updated to 2.0.1 to build with the latest ruby version. So, I guess this is an issue with `rdoc`

**Comment by isuruf** [comment]
Problem is that with ruby 2.4.0-dev, this is checked strictly. Earlier when ruby 2.0.1 was installed it was okay even if rdoc needed < 2.0. Now it is enforced and therefore fails.

**Comment by mohawkjohn** [comment]
Well, I guess rdoc isn't actually necessary for running NMatrix, just for
running rdoc. We can always make the add conditional on Ruby or json
version, if the rdoc people haven't resolved this yet.

**Comment by mohawkjohn** [comment]
Is this the problem?

https://github.com/flori/json/issues/303

It looks like we can just drop the json requirement altogether, yes?

**Comment by isuruf** [comment]
Yes, it is. Dropping the json requirement won't help, since rdoc has json as a requirement.

**Comment by isuruf** [comment]
Dropping the requirement will help ruby versions < 2.4

**Comment by mohawkjohn** [comment]
Looks like we need RDoc 5.

https://github.com/rdoc/rdoc/pull/412

**Comment by mohawkjohn** [comment]
Possibly related to #606?
